040-SDK/100-sql-access.mdx

```diff
@@ -0,0 +1,198 @@
+
+```ts
+const { records: myRecord } = await xata.sql<TeamsRecord>`SELECT * FROM teams`;
+```
+
+```python
+records = xata.sql().query("SELECT * FROM \"teams\"")
+```
+
+```jsonc
+// POST https://{workspace}.{region}.xata.sh/db/{db}:{branch}/sql
+
+{
+  "statement": "SELECT * FROM \"teams\""
+}
+```
+
+</TabbedCode>
+
+The following response is returned:
+
+```json
+{
+  "records": [
+    {
+      "id": "rec_c8hng2h26un90p8sr7k0",
+      "name": "Matrix",
+      "owner": {
+        "id": "myid"
+      },
+      "xata": {
+        "version": 0,
+        "createdAt": "2023-05-15T08:21:31.96526+01:00",
+        "updatedAt": "2023-05-15T21:58:54.072595+01:00"
+      }
+    }
+  ]
+}
+```
+
+### Select a subset of columns from your table
+
+<TabbedCode tabs={['TypeScript', 'Python', 'JSON']}>
+
+```ts
+const { records: myRecord } = await xata.sql<TeamsRecord>`SELECT name, city FROM "Users"`;
+```
+
+```python
+records = xata.sql().query("SELECT name, city FROM \"Users\"")
+```
+
+```jsonc
+// POST https://{workspace}.{region}.xata.sh/db/{db}:{branch}/sql
+
+{
+  "statement": "SELECT name, city FROM \"Users\""
+}
+```
+
+</TabbedCode>
+
+Response:
+
+```json
+{
+  "records": [
+    {
+      "city": "New York",
+      "name": "Keanu Reaves"
+    }
+  ]
+}
+```
+
+### Insert a new record using a parameterized query
+
+<TabbedCode tabs={['TypeScript', 'Python', 'JSON']}>
+```ts
+const { records: myRecord } = await xata.sql<TeamsRecord>`INSERT INTO "Users" (name, email) VALUES ($1, $2)`;
+```
+```python
+records = xata.sql().query(
+  "INSERT INTO \"Users\" (name, email) VALUES ($1, $2)",
+  ["Keanu Reeves", "keanu@example.com"]
+)
+```
+```jsonc
+// POST https://{workspace}.{region}.xata.sh/db/{db}:{branch}/sql
+
+{
+  "statement": "INSERT INTO \"Users\" (name, email) VALUES ($1, $2)",
+  "params": ["Keanu Reeves", "keanu@example.com"]
+}
+```
+</TabbedCode>
+
+### Delete record using a parameterized query
+
+```jsonc
+// POST https://{workspace}.{region}.xata.sh/db/{db}:{branch}/sql
+
+{
+  "statement": "DELETE FROM  \"Users\" WHERE id=$1",
+  "params": ["my-record-id"]
+}
+```
+
+
+## Limitations
+
+* `xata.*` internal fields are read-only.
+* You can only submit one statement per request.
+* Xata returns 1000 records at a time. Please access your data using `LIMIT` and `OFFSET` keywords to iterate over your data.
+* JSON formatter functions do not work as the endpoint already returns a JSON response.
+* Columns with the same name are overwritten by default. If you have multiple columns with the same name please use aliasing to rename at least one of them. For example: 
+  ```sql
+  SELECT address AS physical_address, address AS email 
+  FROM people, emails 
+  WHERE people.id=emails.id;
+  ```
+* Xata does not support DDL commands like CREATE, DROP, ALTER and TRUNCATE.
+* Xata does not support DCL commands like GRANT or REVOKE.
+* Xata does not support TCL commands like COMMIT, ROLLBACK, SAVEPOINT, etc.
```

Comment:
This looks a little awkward in the middle. Code blocks and bullet points don't really mix. I'd suggest adding that part bellow the bullet list. (Sorry I can't do this one as a suggestion, GH doesn't like all the fences)

Something like...

* Bullet list item
* Bullet list item

Columns with the same name are overwritten by default. If you have multiple columns with the same name please use aliasing to rename at least one of them. For example:

```sql
SELECT address AS physical_address, address AS email 
FROM people, emails 
WHERE people.id=emails.id;
```

Reply:
@snide I rearranged the section to move the SQL snippet out of the bullet points list. Similar with the redundant leading bullet points, moved out to it's own list.

@joan-ing we can break SQL further in follow up PRs. I changed the slug to `sdk/sql/overview`. If we move to a sub-folder structure we won't need any redirects hopefully 😄 🤞